### PR TITLE
Consolidating MLN and CG to DL4J

### PR DESCRIPTION
**Pull request by ShamsUlAzeem**
Fixing #214

### CODE REVIEW

File: konduit-serving-core/src/main/java/ai/konduit/serving/executioner/inference/factory/Dl4jInferenceExecutionerFactory.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+            List<String> inputNames = computationGraph2.getConfiguration().getNetworkInputs();
+            List<String> outputNames = computationGraph2.getConfiguration().getNetworkOutputs();
+            log.info("Loaded computation graph with input names " + inputNames + " and output names " + outputNames);
+
+            return InitializedInferenceExecutionerConfig.builder()
+                    .inferenceExecutioner(inferenceExecutioner)
+                    .inputNames(inputNames).outputNames(outputNames)
+                    .build();
+        }
+    }
```
Comment (AlexDBlack) [inline konduit-serving-core/src/main/java/ai/konduit/serving/executioner/inference/factory/Dl4jInferenceExecutionerFactory.java]: I'd probably do
```
if(validateMultiLayerNetwork(...).isValid()){
    ...
} else if(validateComputationGraph...) {
    ...
} else {
    //Throw exception here - "not MLN or CG"
}

```

File: konduit-serving-core/src/main/java/ai/konduit/serving/executioner/inference/factory/Dl4jInferenceExecutionerFactory.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
+package ai.konduit.serving.executioner.inference.factory;
```
Comment (AlexDBlack) [inline konduit-serving-core/src/main/java/ai/konduit/serving/executioner/inference/factory/Dl4jInferenceExecutionerFactory.java]: Copyright header

File: konduit-serving-core/src/main/java/ai/konduit/serving/executioner/inference/factory/Dl4jInferenceExecutionerFactory.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+            ParallelInferenceConfig parallelInferenceConfig = modelPipelineStepConfig.getParallelInferenceConfig();
+
+            ComputationGraphModelLoader computationGraphModelLoader = new ComputationGraphModelLoader(modelPath);
+            MultiComputationGraphInferenceExecutioner inferenceExecutioner = new MultiComputationGraphInferenceExecutioner();
+            inferenceExecutioner.initialize(computationGraphModelLoader, parallelInferenceConfig);
+
+            ComputationGraph computationGraph2 = computationGraphModelLoader.loadModel();
+            List<String> inputNames = computationGraph2.getConfiguration().getNetworkInputs();
+            List<String> outputNames = computationGraph2.getConfiguration().getNetworkOutputs();
+            log.info("Loaded computation graph with input names " + inputNames + " and output names " + outputNames);
```
Comment (AlexDBlack) [inline konduit-serving-core/src/main/java/ai/konduit/serving/executioner/inference/factory/Dl4jInferenceExecutionerFactory.java]: Logging nitpick:
```
log.info("Loaded computation graph with input names {} and output names {}", inputNames, outputNames);
```

**Comment by ShamsUlAzeem** [comment]
@AlexDBlack added the requested changes. I've also put some logic for printing the error messages while importing invalid dl4j models. Btw the tests are gonna pass when we merge the dl4j konduitai master to eclipse master so that the snapshots are updated.

**Comment by AlexDBlack** [comment]
After updating snapshots, I've retriggered CI... Linux is fine, but Windows is stuck on `Waiting for next available executor on ‘windows-x86_64-cpu’` ATM
If/when that passes, this can be merged.

**Comment by ShamsUlAzeem** [comment]
Hmm @AlexDBlack do you see the same thing as me? The CI passes but it doesn't show here on github. Maybe I'm replaying it wrong?

**Comment by AlexDBlack** [comment]
No, Windows CI isn't passing. The new run had some JVM crashes:
[log (2).txt](https://github.com/KonduitAI/konduit-serving/files/4238651/log.2.txt)

I'd say that is unrelated to your changes though...

**Comment by AlexDBlack** [comment]
I've merged latest master and CI has restarted... doubt that will fix it but we'll see

**Comment by AlexDBlack** [comment]
Passing now. So either it's fixed, or it's intermittent 🤷‍♂

**Pull request merged**